ember-server/shared/src/main/scala/org/http4s/ember/server/internal/WebSocketHelpers.scala

```diff
@@ -234,6 +236,49 @@ private[internal] object WebSocketHelpers {
       go(stream, Array.emptyByteArray).void.stream
     }
 
+  private def aggregateFragment[F[_]]: Pipe[F, WebSocketFrame, WebSocketFrame] =
+    stream => {
+      // Takes a chunk of WebSocketFrames and aggregates fragmented frames into one frame.
+      def aggregate(
+          frames: Chunk[WebSocketFrame]
+      ): (Chunk[WebSocketFrame], Chunk[WebSocketFrame]) = {
+        val initialState = (Chunk.empty[WebSocketFrame], Chunk.empty[WebSocketFrame])
+        frames.foldLeft(initialState) {
+          // Current frame is a single frame (not fragmented), or the last one of a sequence of fragments.
+          case ((result, fragments), curFrame) if curFrame.last =>
+            val fragmentSum = fragments ++ Chunk(curFrame)
+            val aggregatedData = fragmentSum.map(_.data).foldLeft(ByteVector.empty)(_ ++ _)
+            val aggregatedFrame = fragmentSum.head.fold(result ++ Chunk(curFrame)) { firstFrame =>
+              firstFrame match {
+                case WebSocketFrame.Text(_, _) =>
+                  result ++ Chunk(WebSocketFrame.Text(aggregatedData, true))
+                case WebSocketFrame.Binary(_, _) =>
+                  result ++ Chunk(WebSocketFrame.Binary(aggregatedData, true))
+                case _: WebSocketFrame =>
```

Comment:
I built something similar when toying around with ember and stumbled over `Continuation` blaze handles them here https://github.com/http4s/blaze/blob/main/blaze-server/src/main/scala/org/http4s/blaze/server/WSFrameAggregator.scala#L54 might need a slight adjustment here to handle them?

Reply:
Thank you for the comment :)

As far as I'm concerned, I think the Continuation case is not necessary here because this section is where `curFrame` is the last fragmented-frame, and the last fragmented Continuation frame should matches the `Text` or `Binary` cases depending on their first fragmented frame type. (Also, a single Continuation frame with fin bit 1 is not considered here, because this case might not be the part of the [specification](https://www.rfc-editor.org/rfc/rfc6455#section-5.4)).

The third match here is intended to handle ControlFrames (such as `Ping` or `Close`) that come in singly.

However, as you said, this area is subtle a bit, so I'll write more detailed tests.